### ISSUE TITLE
Question regarding subcribing to Connectivity.ConnectivityChanged to update state

### ISSUE THREAD

**Issue by MitchBomcanhao** [jsuarezruiz/XamarinFormsStateTriggers#1]
Hi @jsuarezruiz 

You're using this sort of code to subscribe to this event
```
        public NetworkConnectionStateTrigger()
        {
            UpdateState();

            // TODO: Subscribe and unsubscribe this event using OnAttached and OnDetach.
            Connectivity.ConnectivityChanged += OnConnectivityChanged;
        }
```

I've had issues with similar approaches, as the ConnectivityChanged event may not fire or may not be listened to if the app is suspended when the connection state changes.
This means that sometimes the app would be stuck with the wrong connectivity status until something triggered connectivity to recheck itself.
You mention using the attach/detach methods to manage the subscription - what would we need to get the suspend/resume scenarios working reliably?

**Comment by vhugogarcia** [comment]
I'm facing the same question as you @MitchBomcanhao . Did you figured it out?

**Comment by MitchBomcanhao** [comment]
@vhugogarcia we've just started to actively poll for the connectivity state before starting tasks that would perform certain web requests, and take the penalty, if any. our telemetry shows that this still happens quite frequently, ie the app thinks it is offline according to the connectivity state, then we check it and it is actually online, as expected, as the connectivity changed while the app was offscreen. It is a bit ugly but it works.

**Comment by vhugogarcia** [comment]
That is interesting. So before any web request you check for conmectivity maybe by pinga webservice and based on the response you go to next steps?

**Comment by MitchBomcanhao** [comment]
@vhugogarcia

**ok, this is the starting position:**
your device is offline, your app is running and is aware that it is offline (you use the `Connectivity.ConnectivityChanged` event to keep updating the `IsNetworkAvailable` flag in a class called `NetworkConnectivity`).
you send the app into the background and change the device back to online. this change did not trigger the above event and your app still thinks it is offline.

**what we're now doing:**
on the method that does web requests, the first thing we do is check the connectivity flag we're keeping track of.

because it is set to offline, we're going to confirm that it is really offline - to achieve that, we've added a new method to our `NetworkConnectivity `class called `UpdateNetworkAvailability`. All it does is something like this
`IsNetworkAvailable = Connectivity.NetworkAccess == NetworkAccess.Internet;`
The simple use of Connectivity.NetworkAccess forces the Connectivity plugin to do its magic and it'll update its value.

This is far from optimised and we're only doing it when the device thinks it is offline - having it think it is online and then failing on the web requests is harmless for our application.

summing up - all we're doing is getting the  `Connectivity.NetworkAccess ` value just before we run the chunk of code that is doing web requests, to ensure it gets unblocked in case the app wrongly still thinks it is offline.

**Comment by vhugogarcia** [comment]
That makes sense now. Thank you for sharing your approach.
